src/detect-replace.c

```diff
@@ -671,30 +663,20 @@ static int DetectReplaceParseTest03(void)
     int run_mode_backup = run_mode;
     run_mode = RUNMODE_NFQ;
 
-    DetectEngineCtx *de_ctx = NULL;
-    int result = 1;
+    DetectEngineCtx *de_ctx = DetectEngineCtxInit();
 
-    de_ctx = DetectEngineCtxInit();
-    if (de_ctx == NULL)
-        goto end;
+    FAIL_IF_NULL(de_ctx);
 
     de_ctx->flags |= DE_QUIET;
-    de_ctx->sig_list = SigInit(de_ctx,
-                               "alert tcp any any -> any any "
-                               "(msg:\"test\"; content:\"doh\"; replace:\"don\"; http_header; sid:238012;)");
-    if (de_ctx->sig_list != NULL) {
-        result = 0;
-        goto end;
-    }
+    de_ctx->sig_list = DetectEngineAppendSig(de_ctx,
+            "alert tcp any any -> any any "
+            "(msg:\"test\"; content:\"doh\"; replace:\"don\"; http_header; sid:238012;)");
```

Comment:
I realize that I may have led to a mistake. I'm sorry about that.
We use `DetectEngineAppendSig()` when we have a signature declared.

You may follow the approach used in the test in `detect-content.c`, and just call it from within the maccro: https://github.com/OISF/suricata/blob/master/src/detect-content.c#L3084